activesupport/lib/active_support/testing/method_call_assertions.rb

```diff
@@ -35,6 +35,53 @@ def assert_not_called(object, method_name, message = nil, &block)
           assert_called(object, method_name, message, times: 0, &block)
         end
 
+        # TODO: No need to resort to #send once support for Ruby 2.4 is
+        # dropped. May even use prepended module. See:
+        # https://github.com/rails/rails/pull/33162#issuecomment-408616175 .
```

Comment:
There is no need to keep method calls that were only introduced to support a version of Ruby once support for that version is dropped. Unnecessary method calls are not a good idea either for performance code clarity.
I have not checked the other places, but it may be a good idea to check when the time comes.